lib/cog/commands/bundle.ex

```diff
@@ -69,70 +75,130 @@ defmodule Cog.Commands.Bundle do
        disabled.
 
        Cannot be used on the `#{Cog.embedded_bundle}` bundle.
-
   """
 
-  use Cog.Command.GenCommand.Base, bundle: Cog.embedded_bundle
-
-  alias Cog.Repo
-  alias Cog.Models.Bundle
-
   permission "manage_commands"
 
   rule "when command is #{Cog.embedded_bundle}:bundle must have #{Cog.embedded_bundle}:manage_commands"
 
-   def handle_message(req, state) do
-     case run(req.args) do
-       {:ok, response} ->
-         {:reply, req.reply_to, "bundle", response, state}
-       {:error, error} ->
-         {:error, req.reply_to, error_message(error), state}
-     end
-   end
-
-   defp run([action, bundle_name]) when is_binary(bundle_name)
-                                    and action in ["enable", "disable"] do
-     case find_bundle(bundle_name) do
-       {:ok, bundle} ->
-         case Bundle.Status.set(bundle, action_to_status(action)) do
-           {:ok, bundle} ->
-             {:ok, Bundle.Status.current(bundle)}
-           {:error, :embedded_bundle}=error ->
-             error
-         end
-       {:error, _}=error ->
-         error
-     end
-   end
-   defp run(["status", bundle_name]) when is_binary(bundle_name) do
-     case find_bundle(bundle_name) do
-       {:ok, bundle} ->
-         {:ok, Bundle.Status.current(bundle)}
-       {:error, _}=error ->
-         error
-     end
-   end
-   defp run(_) do
-     {:error, :invalid_invocation}
-   end
-
-   defp action_to_status("enable"), do: :enabled
-   defp action_to_status("disable"), do: :disabled
-
-   defp find_bundle(name) do
-     case Repo.get_by(Bundle, name: name) do
-       nil ->
-         {:error, {:not_found, name}}
-       %Bundle{}=bundle ->
-         {:ok, bundle}
-     end
-   end
-
-   defp error_message(:embedded_bundle),
-     do: "The status of the embedded bundle `#{Cog.embedded_bundle}` cannot be changed!"
-   defp error_message({:not_found, name}),
-     do: "The bundle `#{name}` cannot be found!"
-   defp error_message(:invalid_invocation),
-     do: "That is not a valid invocation of the `bundle` command"
-
+  def handle_message(%{args: ["status", bundle_name]} = req, state) do
+    case Bundles.with_status_by_name(bundle_name) do
+      nil ->
+        {:error, req.reply_to, error_message({:not_found, bundle_name}), state}
+      bundle ->
+        {:reply, req.reply_to, "bundle-status", bundle, state}
+    end
+  end
+
+  def handle_message(%{args: [action, "operable"|_]} = req, state)
```

Comment:
This should be handled in the repository function call for you... I don't think you need to special-case it here.

(Incidentally, you shouldn't be able to enable or disable the `site` bundle either.)